options/options.go

```diff
@@ -460,24 +463,33 @@ func (opts *ToolOptions) ParseArgs(args []string) ([]string, error) {
 
 	failpoint.ParseFailpoints(opts.Failpoints)
 
-	err = opts.NormalizeHostPortURI()
+	err = opts.NormalizeOptionsAndURI()
 	if err != nil {
 		return []string{}, err
 	}
 
 	return args, err
 }
 
-func (opts *ToolOptions) NormalizeHostPortURI() error {
+// NormalizeOptionsAndURI syncs the connection string an toolOptions objects.
+// It returns an error if there is any conflict between options and the connection string.
+// If a value is set on the options, but not the connection string, that value is added to the
+// connection string. If a value is set on the connection string, but not the options,
+// that value is added to the options.
+func (opts *ToolOptions) NormalizeOptionsAndURI() error {
 	if opts.URI != nil && opts.URI.ConnectionString != "" {
-		cs, err := connstring.Parse(opts.URI.ConnectionString)
+		cs, err := connstring.ParseWithoutValidating(opts.URI.ConnectionString)
```

Comment:
This actually would have helped to give better error messages in one of the Mirror tickets I did, nice!